src/rand/rpo.rs

```diff
@@ -174,6 +174,36 @@ impl RandomCoin for RpoRandomCoin {
 
         Ok(values)
     }
+
+    fn reseed_with_salt(
+        &mut self,
+        data: <Self::Hasher as winter_crypto::Hasher>::Digest,
+        salt: Option<<Self::Hasher as winter_crypto::Hasher>::Digest>,
+    ) {
```

Comment:
What is this method used for? Also, we should probably add some doc comments to it.

Reply:
This is a method that reseeds using a random salt. This is needed for the zk of the BCS transformed protocol.
I think that we should rename this to `reseed` and remove the old one but we should discuss this in the relevant PR in Winterfell.
As for the comment, I think we should do it at the trait definition level but open to adding it here too